### ISSUE TITLE
endpoint for experimental installer

### ISSUE THREAD

**Issue by abathur** [NixOS/nixos-homepage#1614]
Hi @NixOS/marketing-team,

Not sure what we'll need to close the gap(s) here, but we hope to get releases of https://github.com/NixOS/experimental-nix-installer hosted under nixos.org. (Not replacing existing installer endpoints--it should clearly be an experiment.)

There's background in the infra issue we opened earlier this year:
- https://github.com/NixOS/infra/issues/452

Reaching out as the infra team suggested in issue above now that we have a release workflow set up.

The [current tentative prerelease](https://github.com/NixOS/experimental-nix-installer/releases/tag/0.27.0) should be accessible via something like:
```shell
curl --proto '=https' --tlsv1.2 -sSf -L https://github.com/NixOS/experimental-nix-installer/releases/download/0.27.0/nix-installer.sh | sh -s -- --help
```

I'm not sure how ~controlled the sanctioning of a latest release should be, but I did also briefly un-toggle the prerelease status in order to confirm the latest release per github would also be accessible as:
```shell
curl --proto '=https' --tlsv1.2 -sSf -L https://github.com/NixOS/experimental-nix-installer/releases/latest/download/nix-installer.sh | sh -s -- --help
```

(I toggled it back to prerelease since we've already had people ~depend on this installer as we work, making me skittish about leaving a normal release sitting around for unknown amount of time without a clear go-ahead.)

cc @tomberek @roberth @mkenigs

**Comment by Mic92** [comment]
Ok. Will see what it takes to make it happen.

**Comment by Mic92** [comment]
@abathur should we just make this a redirect on the homepage?

**Comment by abathur** [comment]
@Mic92 Jonas suggested a redirect in https://github.com/NixOS/infra/issues/452#issuecomment-2226855264, and I'm inclined to defer to anyone with a clear opinion on it for now :)

(I'm unsure if a redirect is how we'd want to handle this permanently if adopted as the official installer, but I think it is as good a place to start as any. I suspect it'll be easier to have a better version of this conversation nearer official adoption when more stakeholders are up to speed, if that makes sense.)

(also cc @idabzo @mkenigs @tomberek since we discussed this about an hour ago in the installer WG meeting)

**Comment by Mic92** [comment]
Ok. I finished my research. First we need a redirect on the homepage:

This is just a file here:

```
https://github.com/NixOS/nixos-homepage/blob/f24bea80eebb99c1bc8c6ca17b6b9140acb75165/public/netlify.toml#L447C11-L447C24
```


Than I would create you a new s3 bucket where you can upload releases. Maybe actually put fastly in front because our current nix release bucket is directly exposed to the internet. If we make the experimental installer the new default we can take the fastly configuration and put it front of our current s3 release bucket as well and the installer would be directly uploaded their instead of the current nix releases.

**Comment by Mic92** [comment]
Instead of handing out s3 credentials, I could also use the opportunity to make the github repository in github actions authenticate against the release bucket using oicd. Something I meant to do for the new normal nix repository. That would also simplify the release process in future.

**Comment by mkenigs** [comment]
@Mic92 do you prefer that approach of uploading to s3 rather than just adding a redirect? I think we're fine with either, just want to make sure that if it's easier to just do a redirect you know we're fine with that too

**Comment by Mic92** [comment]
The idea also came up to just have fastly as a proxy in front of github releases.
I just don't want github only because it doesn't do ipv6 also having some indirection allows to move to something else if needed. Fastly also had better uptime so far than github.

**Comment by thilobillerbeck** [comment]
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/2548365?s=64&amp;u=a9ab22e8b0e0089dc386633ed1fc292ba04e8d03&amp;v=4">@abathur](https://github.com/abathur) should we just make this a redirect on the homepage to github or do you want to upload this to NixOS own homepage?

I guess a redirect would be better and would cause less interference

**Comment by thilobillerbeck** [comment]
Btw. just ping me if you need any help or advice. I interpreted this issue more like a self containing thing that worked on its own or a note, since getting things hosted under nixos.org is a rather loose term and does not directly mean hosting on the nixos homepage, which btw. is something we should avoid (and just use redirects).

**Comment by abathur** [comment]
> getting things hosted under nixos.org is a rather loose term

Yes--intentionally just asking for a broad outcome (the experimental installer is available via nixos.org or some subdomain thereof). 

I'm not sure what's ideal (especially since these might differ between the clearly-marked experiment stage and final adoption) and want to avoid misguiding that discussion by being too specific.

**Comment by abathur** [comment]
@thilobillerbeck I don't think this is settled as the marketing team minutes above suggest, though I'm less clear on whose court the ball is in and who the deciders are.

I think there's good agreement on a redirect, but less on where to :)

@Mic92 Are you blocked on ~keys to the installer repo? I recall you asking about it, but my own permissions aren't sufficient to add you. I can ask via the org owners helpdesk if we're stuck there.

**Comment by Mic92** [comment]
No. Sorry. I was just blocked on time. Had to finish something before a doing a longer trip. But this task is still on the back of my mind.

**Comment by thilobillerbeck** [comment]
> @thilobillerbeck I don't think this is settled as the marketing team minutes above suggest, though I'm less clear on whose court the ball is in and who the deciders are.

Sorry, the meeting notes may have been a bit unclear there. What was resolved was your outreach/ notice to us regarding this issue (which I then offered you help in) not the issue or our involvement itself. :D

**Comment by abathur** [comment]
@Mic92 Just want to check back in on this since Valentin queried about where it stands :)

**Comment by fricklerhandwerk** [comment]
CC @erethon @imincik since having an official Rust installer may improve the NGIpkgs UX.

**Comment by Mic92** [comment]
Would be great if someone could help me providing the fastly configuration similiar to how this was handled in this pull request: https://github.com/NixOS/infra/pull/562
Configuration can be tested with a free fastly account.

**Comment by tomberek** [comment]
@Mic92 Who has the knowledge to help here? If no one, then who can step in to do it?

**Comment by Mic92** [comment]
gh-releases.nixos.org is now online. We will rename this to assets.nixos.org but than we are good.

**Comment by tomberek** [comment]
What endpoints are valid at that host?

**Comment by mkenigs** [comment]
@Mic92 so I assume we'll need to auth to gh-releases.nixos.org in a GH action in https://github.com/NixOS/experimental-nix-installer, how do you want us to set up creds?
